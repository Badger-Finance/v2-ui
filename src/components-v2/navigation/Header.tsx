import React from 'react';
import { observer } from 'mobx-react-lite';
import views from '../../config/routes';
import { useContext } from 'react';
import { StoreContext } from '../../mobx/store-context';
import { Toolbar, AppBar, IconButton } from '@material-ui/core';
import { Menu } from '@material-ui/icons';
import { useStyles } from './HeaderStyle';
import { NotifyHook } from './hooks/NotifyHook';

export const Header = observer(() => {
	const classes = useStyles();

	const store = useContext(StoreContext);
	const {
		router: { goTo },
		uiState: { openSidebar },
	} = store;

	//call Notify Hook
	NotifyHook()

	return (
		<AppBar className={classes.appBar} color="primary">
			<Toolbar className={classes.toolbar} onClick={() => goTo(views.home, { collection: 'badger' })}>
				<IconButton className={classes.menuButton} onClick={() => openSidebar()}>
					<Menu />
				</IconButton>
				<img alt="" src={require('../../assets/badger-full.png')} className={classes.logo} />
			</Toolbar>
		</AppBar>
	);
});
