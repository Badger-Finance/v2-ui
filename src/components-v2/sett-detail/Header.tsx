import React from 'react';
import { Grid, Link, makeStyles } from '@material-ui/core';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import GasWidget from '../common/GasWidget';
import WalletWidget from '../common/WalletWidget';
import { HeaderContainer } from '../common/Containers';
import { StoreContext } from '../../mobx/store-context';
import routes from '../../config/routes';
import SettStore from 'mobx/stores/SettStore';
import { SettState } from 'mobx/model/setts/sett-state';

const useStyles = makeStyles((theme) => ({
	root: {
		[theme.breakpoints.down('sm')]: {
			marginTop: theme.spacing(8),
		},
	},
	widgets: {
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'flex-end',
		[theme.breakpoints.down('xs')]: {
			marginTop: theme.spacing(3),
			justifyContent: 'center',
		},
	},
	backArrow: {
		marginRight: 12,
	},
	links: {
		display: 'flex',
		alignItems: 'center',
	},
}));

const getRouteBySlug = (slug: string | undefined, setts: SettStore) => {
	const sett = slug ? setts.getSettBySlug(slug) : null;
	if (!slug || !sett) return routes.home;

	switch (sett.state) {
		case SettState.Guarded:
			return routes.guarded;
		case SettState.Experimental:
			return routes.experimental;
		default:
			return routes.home;
	}
};

export const Header = (): JSX.Element => {
	const { router, setts } = React.useContext(StoreContext);
	const classes = useStyles();
	const settSlug = router.params?.settName?.toString();

	return (
		<HeaderContainer container className={classes.root}>
			<Grid item xs={12} sm={6}>
				<Link
					component="button"
					className={classes.links}
					onClick={() => router.goTo(getRouteBySlug(settSlug, setts))}
				>
					<ArrowBackIcon className={classes.backArrow} />
					Back to All Setts
				</Link>
			</Grid>
			<Grid item xs={12} sm={6} className={classes.widgets}>
				<GasWidget />
				<WalletWidget />
			</Grid>
		</HeaderContainer>
	);
};
