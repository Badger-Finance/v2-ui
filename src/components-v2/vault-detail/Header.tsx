import { Link, makeStyles } from '@material-ui/core';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import { StoreContext } from 'mobx/stores/store-context';
import React from 'react';

import routes from '../../config/routes';
import { PageHeaderContainer } from '../common/Containers';

const useStyles = makeStyles((theme) => ({
	root: {
		marginTop: theme.spacing(1),
		marginBottom: theme.spacing(3),
	},
	widgets: {
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'flex-end',
		[theme.breakpoints.down('xs')]: {
			marginTop: theme.spacing(3),
			justifyContent: 'center',
		},
	},
	backArrow: {
		marginRight: 12,
	},
	links: {
		display: 'flex',
		alignItems: 'center',
	},
}));

export const Header = (): JSX.Element => {
	const { router } = React.useContext(StoreContext);
	const classes = useStyles();

	return (
		<PageHeaderContainer container className={classes.root}>
			<Link component="button" className={classes.links} onClick={() => router.goTo(routes.home)}>
				<ArrowBackIcon className={classes.backArrow} />
				Back to All Vaults
			</Link>
		</PageHeaderContainer>
	);
};
