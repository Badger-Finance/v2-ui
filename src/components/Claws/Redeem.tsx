import React, { FC, useState } from 'react';
import { Box, Button, Grid, InputBase, makeStyles, Typography } from '@material-ui/core';
import ClawParams, { ClawParam } from './ClawParams';
import { useMainStyles } from './index';
import ClawLabel from './ClawLabel';
import ClawDetails from './ClawDetails';

const options = ['eCLAW FEB29', 'eCLAW MARCH29'];

const eCLAWS: Record<string, string> = {
	'eCLAW FEB29': '1000',
	'eCLAW MARCH29': '2000',
};

const redeemableTokens: Record<string, string> = {
	'eCLAW FEB29': 'wBTCwETHSLP',
	'eCLAW MARCH29': 'bBadger',
};

const initialValue = { amount: '0.00' };

const useStyles = makeStyles((theme) => ({
	border: {
		border: '1px solid #5C5C5C',
		borderRadius: 8,
	},
	selectContainer: {
		[theme.breakpoints.only('xs')]: {
			justifyContent: 'space-between',
		},
		[theme.breakpoints.up('lg')]: {
			paddingLeft: '10%',
		},
	},
	margin: {
		margin: theme.spacing(1),
	},
	centered: {
		margin: 'auto',
	},
}));

const Redeem: FC = () => {
	const mainClasses = useMainStyles();
	const classes = useStyles();
	const [redeemParams, setRedeemParams] = useState<ClawParam>(initialValue);

	return (
		<Grid container>
			<Box clone pb={2}>
				<Grid item xs={12}>
					<Box clone pb={1}>
						<Grid item xs={12}>
							<ClawLabel
								name="Token"
								balanceLabel={
									redeemParams.selectedOption && `Available ${redeemParams.selectedOption}:`
								}
								balance={redeemParams.selectedOption && eCLAWS[redeemParams.selectedOption]}
							/>
						</Grid>
					</Box>
					<ClawParams
						referenceBalance={redeemParams.selectedOption && eCLAWS[redeemParams.selectedOption]}
						placeholder="Select Token"
						amount={redeemParams.amount}
						onAmountChange={(amount: string, error?: boolean) => {
							setRedeemParams({
								...redeemParams,
								amount,
								error: error ? `Amount exceeds ${redeemParams.selectedOption} balance` : undefined,
							});
						}}
						selectedOption={redeemParams.selectedOption}
						onOptionChange={(selectedOption: string) => {
							setRedeemParams({
								selectedOption,
								amount: '0.00',
								error: undefined,
							});
						}}
						disabledAmount={!redeemParams.selectedOption}
					/>
				</Grid>
			</Box>
			{redeemParams.selectedOption && (
				<Box clone py={2}>
					<Grid item xs={12} sm={8} className={classes.centered}>
						<Box clone pb={1}>
							<Grid item xs={12}>
								<ClawLabel name="You Receive" />
							</Grid>
						</Box>
						<Box clone py={1} px={2}>
							<Grid container alignContent="center" alignItems="center" className={classes.border}>
								<Grid item xs={12}>
									<Grid container alignItems="center" spacing={2} className={classes.selectContainer}>
										<Grid item xs={12} sm={6}>
											<Typography>{redeemableTokens[redeemParams.selectedOption]}</Typography>
										</Grid>
										<Grid item xs={12} sm={6}>
											<InputBase type="tel" disabled placeholder="0.00" value="" />
										</Grid>
									</Grid>
								</Grid>
							</Grid>
						</Box>
					</Grid>
				</Box>
			)}
			<Grid item xs={12}>
				<Grid container className={mainClasses.details}>
					<ClawDetails
						details={[
							{ 'Expiration Date': 'Feb 29, 2021 8:00 UTC' },
							{ 'Expiration Price': '1 eCLAW FEB29 = .000001 wBTCWethSLP' },
						]}
					/>
				</Grid>
			</Grid>
			<Grid item xs={12}>
				<Grid container>
					<Button
						color="primary"
						variant="contained"
						disabled={!!redeemParams.error || !redeemParams.selectedOption}
						size="large"
						className={mainClasses.button}
					>
						{redeemParams.error ? redeemParams.error : 'REDEEM'}
					</Button>
				</Grid>
			</Grid>
		</Grid>
	);
};

export default Redeem;
