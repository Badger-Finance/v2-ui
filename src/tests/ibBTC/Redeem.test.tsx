import React from 'react';
import '@testing-library/jest-dom';
import addresses from 'config/ibBTC/addresses.json';
import store from 'mobx/RootStore';
import { StoreProvider } from '../../mobx/store-context';
import { customRender, screen, fireEvent, act } from '../Utils';
import { Redeem } from '../../components/IbBTC/Redeem';
import { IbbtcOptionToken } from '../../mobx/model/tokens/ibbtc-option-token';
import Header from '../../components/Header';
import { Snackbar } from '../../components/Snackbar';

const tokensConfig = addresses.mainnet.tokens;

describe('ibBTC Redeem', () => {
	beforeEach(() => {
		store.ibBTCStore.ibBTC = new IbbtcOptionToken(store, tokensConfig['ibBTC']);
		store.ibBTCStore.tokens = [new IbbtcOptionToken(store, addresses.mainnet.tokens['bcrvRenBTC'])];
		store.ibBTCStore.ibBTC.balance = store.ibBTCStore.ibBTC.scale('5');
		store.ibBTCStore.calcRedeemAmount = jest.fn().mockReturnValue({
			fee: store.ibBTCStore.ibBTC.scale('0.0120'),
			max: store.ibBTCStore.ibBTC.scale('15'),
			sett: store.ibBTCStore.tokens[0].scale('11.988'),
		});
		store.ibBTCStore.getRedeemConversionRate = jest.fn().mockReturnValue(store.ibBTCStore.tokens[0].scale('1'));
		store.ibBTCStore.redeem = jest.fn();
	});

	it('displays ibBTC balance and output token balance', () => {
		store.ibBTCStore.ibBTC.balance = store.ibBTCStore.ibBTC.scale('10');
		customRender(
			<StoreProvider value={store}>
				<Redeem />
			</StoreProvider>,
		);

		expect(screen.getByText('Balance: 10.000000')).toBeInTheDocument();
	});

	it('can apply max balance', async () => {
		store.ibBTCStore.ibBTC.balance = store.ibBTCStore.ibBTC.scale('5');
		const { container } = customRender(
			<StoreProvider value={store}>
				<Redeem />
			</StoreProvider>,
		);
		await act(async () => {
			await fireEvent.click(await screen.findByRole('button', { name: /max/i }));
		});
		expect(container).toMatchSnapshot();
	});

	it('handles not connected wallet', () => {
		customRender(
			<StoreProvider value={store}>
				<Redeem />
			</StoreProvider>,
		);

		expect(screen.getByRole('textbox')).toBeDisabled();
		expect(screen.getByRole('button', { name: 'Connect Wallet' })).toBeEnabled();
	});

	describe('Input Change', () => {
		beforeEach(() => {
			jest.useFakeTimers();

			store.onboard.address = '0x1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a';
			store.ibBTCStore.calcRedeemAmount = jest.fn().mockReturnValue({
				fee: store.ibBTCStore.ibBTC.scale('0.0120'),
				max: store.ibBTCStore.ibBTC.scale('100'),
				sett: store.ibBTCStore.tokens[0].scale('20'),
			});
		});

		it('displays output balance when redeem amount is inputted', async () => {
			const { container } = customRender(
				<StoreProvider value={store}>
					<Redeem />
				</StoreProvider>,
			);

			fireEvent.change(await screen.findByRole('textbox'), { target: { value: '12' } });

			await screen.findByText('20.000000 bcrvRenBTC');

			expect(container).toMatchSnapshot();
		});

		it('handles exceeding ibBTC redeem input amount', async () => {
			customRender(
				<StoreProvider value={store}>
					<Snackbar>
						<Header />
						<Redeem />
					</Snackbar>
				</StoreProvider>,
			);

			fireEvent.change(screen.getByRole('textbox'), { target: { value: '20' } });

			await screen.findByText('20.000000 bcrvRenBTC');

			fireEvent.click(screen.getByRole('button', { name: /redeem/i }));

			expect(screen.getByText('You have insufficient balance of ibBTC')).toBeInTheDocument();
		});

		it('executes redeem with correct params', async () => {
			const redeemSpy = jest.fn();
			store.ibBTCStore.redeem = redeemSpy;

			customRender(
				<StoreProvider value={store}>
					<Snackbar>
						<Header />
						<Redeem />
					</Snackbar>
				</StoreProvider>,
			);

			fireEvent.change(screen.getByRole('textbox'), { target: { value: '0.1' } });

			await screen.findByText('20.000000 bcrvRenBTC');

			fireEvent.click(screen.getByRole('button', { name: /redeem/i }));

			await screen.findByDisplayValue('');

			expect(redeemSpy).toHaveBeenNthCalledWith(
				1,
				store.ibBTCStore.tokens[0],
				store.ibBTCStore.tokens[0].scale('0.1'),
			);
		});

		it('executes calcRedeem and getRedeemConversionRate with correct params', async () => {
			const calcRedeemSpy = jest.fn().mockReturnValue({
				fee: store.ibBTCStore.ibBTC.scale('0.0120'),
				max: store.ibBTCStore.ibBTC.scale('100'),
				sett: store.ibBTCStore.tokens[0].scale('20'),
			});

			const getConversionSpy = jest.fn().mockReturnValue(Promise.resolve(store.ibBTCStore.tokens[0].scale('20')));

			store.ibBTCStore.calcRedeemAmount = calcRedeemSpy;
			store.ibBTCStore.getRedeemConversionRate = getConversionSpy;

			customRender(
				<StoreProvider value={store}>
					<Snackbar>
						<Header />
						<Redeem />
					</Snackbar>
				</StoreProvider>,
			);

			fireEvent.change(await screen.findByRole('textbox'), { target: { value: '12' } });

			await screen.findByText('20.000000 bcrvRenBTC');

			expect(calcRedeemSpy).toHaveBeenNthCalledWith(
				1,
				store.ibBTCStore.tokens[0],
				store.ibBTCStore.ibBTC.scale('12'),
			);

			expect(getConversionSpy).toHaveBeenNthCalledWith(1, store.ibBTCStore.tokens[0]);
		});
	});
});
