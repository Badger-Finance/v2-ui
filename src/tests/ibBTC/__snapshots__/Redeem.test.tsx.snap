// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ibBTC Redeem can apply max balance 1`] = `
<div>
  <div
    class="MuiGrid-root-3 MuiGrid-container-4"
  >
    <div
      class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-106 WithStyles(ForwardRef(Grid))-root-107 MuiGrid-item-5 MuiGrid-grid-xs-12-49"
    >
      <p
        class="MuiTypography-root-109 WithStyles(ForwardRef(Typography))-root-108 MuiTypography-body1-111 MuiTypography-colorTextSecondary-135"
      >
        Balance: 
        5.000
      </p>
    </div>
    <div
      class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-139 WithStyles(ForwardRef(Grid))-root-140 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-grid-xs-12-49"
    >
      <div
        class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-8-45 MuiGrid-grid-sm-7-58"
      >
        <div
          class="MuiInputBase-root-142 WithStyles(ForwardRef(InputBase))-root-141 MuiInputBase-disabled-145"
        >
          <input
            autocomplete="off"
            class="MuiInputBase-input-153 MuiInputBase-disabled-145"
            disabled=""
            inputmode="decimal"
            maxlength="79"
            minlength="1"
            pattern="^[0-9]*[.,]?[0-9]*$"
            placeholder="0.000"
            spellcheck="false"
            type="text"
            value="5.000000000000000000"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-160 WithStyles(ForwardRef(Grid))-root-161 MuiGrid-container-4 MuiGrid-spacing-xs-1-26 MuiGrid-item-5 MuiGrid-grid-xs-4-41 MuiGrid-grid-sm-5-56"
      >
        <div
          class="MuiGrid-root-3 MuiGrid-item-5"
        >
          <button
            class="MuiButtonBase-root-191 MuiButton-root-162 MuiButton-outlined-167 MuiButton-outlinedSizeSmall-179 MuiButton-sizeSmall-183"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-163"
            >
              max
            </span>
            <span
              class="MuiTouchRipple-root-229"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root-3 MuiGrid-item-5"
        >
          <div
            style="align-items: center; display: flex; flex-wrap: nowrap; overflow: hidden;"
          >
            <img
              alt="Interest-Bearing BTC"
              src="/assets/icons/ibbtc.svg"
              style="height: 2rem; margin-right: .2rem; display: block;"
            />
            <div
              class="MuiTypography-root-109 MuiTypography-body1-111"
            >
              ibBTC
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root-3 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-align-items-xs-center-12 MuiGrid-grid-xs-12-49"
  >
    <svg
      class="makeStyles-arrowDown-194"
      fill="none"
      height="18"
      viewBox="0 0 13 18"
      width="13"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M6.5 15.2138L11.6295 10L13 11.3931L6.5 18L-6.08938e-08 11.3931L1.37054 10L6.5 15.2138Z"
        fill="white"
        fill-rule="evenodd"
      />
      <line
        stroke="white"
        stroke-width="2"
        x1="6.5"
        x2="6.5"
        y1="16"
        y2="4.37114e-08"
      />
    </svg>
  </div>
  <div
    class="MuiGrid-root-3 makeStyles-outputContent-1 MuiGrid-container-4"
  >
    <div
      class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-195 WithStyles(ForwardRef(Grid))-root-196 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-grid-xs-12-49"
    >
      <div
        class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-12-49 MuiGrid-grid-sm-7-58 MuiGrid-grid-md-12-77 MuiGrid-grid-lg-7-86"
      >
        <h1
          class="MuiTypography-root-109 WithStyles(ForwardRef(Typography))-root-197 WithStyles(ForwardRef(Typography))-root-198 MuiTypography-h1-114"
        >
          11.988000
        </h1>
      </div>
      <div
        class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-199 WithStyles(ForwardRef(Grid))-root-200 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-grid-xs-12-49 MuiGrid-grid-sm-5-56 MuiGrid-grid-md-12-77 MuiGrid-grid-lg-5-84"
      >
        <button
          class="MuiButtonBase-root-191 MuiButton-root-162 MuiButton-outlined-167 makeStyles-selectButton-206 MuiButton-outlinedSizeSmall-179 MuiButton-sizeSmall-183"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-163"
          >
            <div
              style="align-items: center; display: flex; flex-wrap: nowrap; overflow: hidden;"
            >
              <img
                alt="Badger crvRenWSBTC"
                src="/assets/icons/bcrvrenwsbtc.svg"
                style="height: 2rem; margin-right: .2rem; display: block;"
              />
              <div
                class="MuiTypography-root-109 MuiTypography-body1-111"
              >
                bcrvRenWSBTC
              </div>
            </div>
            <span
              class="MuiButton-endIcon-187 MuiButton-iconSizeSmall-188"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-208"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTouchRipple-root-229"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-12-49"
  >
    <div
      class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-217"
    >
      <div
        class="MuiGrid-root-3 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-justify-xs-space-between-23 MuiGrid-grid-xs-12-49"
      >
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <h6
            class="MuiTypography-root-109 MuiTypography-subtitle1-120"
          >
            Current Conversion Rate: 
          </h6>
        </div>
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <p
            class="MuiTypography-root-109 WithStyles(ForwardRef(Typography))-root-108 MuiTypography-body1-111"
          >
            1 
            ibBTC
             : 
            1.000000
             
            bcrvRenWSBTC
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root-3 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-justify-xs-space-between-23 MuiGrid-grid-xs-12-49"
      >
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <h6
            class="MuiTypography-root-109 MuiTypography-subtitle1-120"
          >
            Fees: 
          </h6>
        </div>
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <p
            class="MuiTypography-root-109 WithStyles(ForwardRef(Typography))-root-108 MuiTypography-body1-111"
          >
            <span
              class=""
              title="Redeem Fee: 0%"
            >
              0.012000
               
              ibBTC
            </span>
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root-3 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-justify-xs-space-between-23 MuiGrid-grid-xs-12-49"
      >
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <h6
            class="MuiTypography-root-109 MuiTypography-subtitle1-120"
          >
            Total Redeem Amount: 
          </h6>
        </div>
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <p
            class="MuiTypography-root-109 WithStyles(ForwardRef(Typography))-root-108 MuiTypography-body1-111"
          >
            11.988000 bcrvRenWSBTC
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-12-49"
  >
    <button
      class="MuiButtonBase-root-191 MuiButton-root-162 MuiButton-contained-170 MuiButton-containedPrimary-171 MuiButton-containedSizeLarge-182 MuiButton-sizeLarge-184 MuiButton-fullWidth-185"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-163"
      >
        Connect Wallet
      </span>
      <span
        class="MuiTouchRipple-root-229"
      />
    </button>
  </div>
</div>
`;

exports[`ibBTC Redeem displays output balance when redeem amount is inputted 1`] = `
<div>
  <div
    class="MuiGrid-root-3 MuiGrid-container-4"
  >
    <div
      class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-106 WithStyles(ForwardRef(Grid))-root-107 MuiGrid-item-5 MuiGrid-grid-xs-12-49"
    >
      <p
        class="MuiTypography-root-109 WithStyles(ForwardRef(Typography))-root-108 MuiTypography-body1-111 MuiTypography-colorTextSecondary-135"
      >
        Balance: 
        0.000
      </p>
    </div>
    <div
      class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-139 WithStyles(ForwardRef(Grid))-root-140 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-grid-xs-12-49"
    >
      <div
        class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-8-45 MuiGrid-grid-sm-7-58"
      >
        <div
          class="MuiInputBase-root-142 WithStyles(ForwardRef(InputBase))-root-141 MuiInputBase-disabled-145"
        >
          <input
            autocomplete="off"
            class="MuiInputBase-input-153 MuiInputBase-disabled-145"
            disabled=""
            inputmode="decimal"
            maxlength="79"
            minlength="1"
            pattern="^[0-9]*[.,]?[0-9]*$"
            placeholder="0.000"
            spellcheck="false"
            type="text"
            value="12"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-160 WithStyles(ForwardRef(Grid))-root-161 MuiGrid-container-4 MuiGrid-spacing-xs-1-26 MuiGrid-item-5 MuiGrid-grid-xs-4-41 MuiGrid-grid-sm-5-56"
      >
        <div
          class="MuiGrid-root-3 MuiGrid-item-5"
        >
          <button
            class="MuiButtonBase-root-191 MuiButton-root-162 MuiButton-outlined-167 MuiButton-outlinedSizeSmall-179 MuiButton-sizeSmall-183"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-163"
            >
              max
            </span>
            <span
              class="MuiTouchRipple-root-229"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root-3 MuiGrid-item-5"
        >
          <div
            style="align-items: center; display: flex; flex-wrap: nowrap; overflow: hidden;"
          >
            <img
              alt="Interest-Bearing BTC"
              src="/assets/icons/ibbtc.svg"
              style="height: 2rem; margin-right: .2rem; display: block;"
            />
            <div
              class="MuiTypography-root-109 MuiTypography-body1-111"
            >
              ibBTC
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root-3 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-align-items-xs-center-12 MuiGrid-grid-xs-12-49"
  >
    <svg
      class="makeStyles-arrowDown-194"
      fill="none"
      height="18"
      viewBox="0 0 13 18"
      width="13"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M6.5 15.2138L11.6295 10L13 11.3931L6.5 18L-6.08938e-08 11.3931L1.37054 10L6.5 15.2138Z"
        fill="white"
        fill-rule="evenodd"
      />
      <line
        stroke="white"
        stroke-width="2"
        x1="6.5"
        x2="6.5"
        y1="16"
        y2="4.37114e-08"
      />
    </svg>
  </div>
  <div
    class="MuiGrid-root-3 makeStyles-outputContent-1 MuiGrid-container-4"
  >
    <div
      class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-195 WithStyles(ForwardRef(Grid))-root-196 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-grid-xs-12-49"
    >
      <div
        class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-12-49 MuiGrid-grid-sm-7-58 MuiGrid-grid-md-12-77 MuiGrid-grid-lg-7-86"
      >
        <h1
          class="MuiTypography-root-109 WithStyles(ForwardRef(Typography))-root-197 WithStyles(ForwardRef(Typography))-root-198 MuiTypography-h1-114"
        >
          11.988000
        </h1>
      </div>
      <div
        class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-199 WithStyles(ForwardRef(Grid))-root-200 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-grid-xs-12-49 MuiGrid-grid-sm-5-56 MuiGrid-grid-md-12-77 MuiGrid-grid-lg-5-84"
      >
        <button
          class="MuiButtonBase-root-191 MuiButton-root-162 MuiButton-outlined-167 makeStyles-selectButton-206 MuiButton-outlinedSizeSmall-179 MuiButton-sizeSmall-183"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-163"
          >
            <div
              style="align-items: center; display: flex; flex-wrap: nowrap; overflow: hidden;"
            >
              <img
                alt="Badger crvRenWSBTC"
                src="/assets/icons/bcrvrenwsbtc.svg"
                style="height: 2rem; margin-right: .2rem; display: block;"
              />
              <div
                class="MuiTypography-root-109 MuiTypography-body1-111"
              >
                bcrvRenWSBTC
              </div>
            </div>
            <span
              class="MuiButton-endIcon-187 MuiButton-iconSizeSmall-188"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root-208"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTouchRipple-root-229"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-12-49"
  >
    <div
      class="MuiGrid-root-3 WithStyles(ForwardRef(Grid))-root-217"
    >
      <div
        class="MuiGrid-root-3 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-justify-xs-space-between-23 MuiGrid-grid-xs-12-49"
      >
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <h6
            class="MuiTypography-root-109 MuiTypography-subtitle1-120"
          >
            Current Conversion Rate: 
          </h6>
        </div>
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <p
            class="MuiTypography-root-109 WithStyles(ForwardRef(Typography))-root-108 MuiTypography-body1-111"
          >
            1 
            ibBTC
             : 
            1.000000
             
            bcrvRenWSBTC
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root-3 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-justify-xs-space-between-23 MuiGrid-grid-xs-12-49"
      >
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <h6
            class="MuiTypography-root-109 MuiTypography-subtitle1-120"
          >
            Fees: 
          </h6>
        </div>
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <p
            class="MuiTypography-root-109 WithStyles(ForwardRef(Typography))-root-108 MuiTypography-body1-111"
          >
            <span
              class=""
              title="Redeem Fee: 0%"
            >
              0.012000
               
              ibBTC
            </span>
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root-3 MuiGrid-container-4 MuiGrid-item-5 MuiGrid-justify-xs-space-between-23 MuiGrid-grid-xs-12-49"
      >
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <h6
            class="MuiTypography-root-109 MuiTypography-subtitle1-120"
          >
            Total Redeem Amount: 
          </h6>
        </div>
        <div
          class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-6-43"
        >
          <p
            class="MuiTypography-root-109 WithStyles(ForwardRef(Typography))-root-108 MuiTypography-body1-111"
          >
            11.988000 bcrvRenWSBTC
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root-3 MuiGrid-item-5 MuiGrid-grid-xs-12-49"
  >
    <button
      class="MuiButtonBase-root-191 MuiButton-root-162 MuiButton-contained-170 MuiButton-containedPrimary-171 MuiButton-containedSizeLarge-182 MuiButton-sizeLarge-184 MuiButton-fullWidth-185"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label-163"
      >
        Connect Wallet
      </span>
      <span
        class="MuiTouchRipple-root-229"
      />
    </button>
  </div>
</div>
`;
